generator/src/generator.rs

```diff
@@ -0,0 +1,767 @@
+                }
+            }
+            // Transfer
+            1 => {
+                let params = TransferParams::new(
+                    [Transfer {
+                        // The token id of wCCD is 0.
+                        token_id: TokenId::new(vec![])?,
+                        amount:   TokenAmount::from(1u32),
+                        from:     Address::Account(self.args.keys.address),
+                        to:       Receiver::Account(self.args.keys.address),
+                        data:     AdditionalData::new(vec![])?,
+                    }]
+                    .to_vec(),
+                )?;
+
+                UpdateContractPayload {
+                    amount:       Amount::zero(),
+                    address:      self.contract_address,
+                    receive_name: OwnedReceiveName::new("cis2_wCCD.transfer".into())?,
+                    message:      OwnedParameter::from_serial(&params)?,
+                }
+            }
+            // Unwrap
+            2 => {
+                let params = UnwrapParams {
+                    amount:   TokenAmount::from(1u32),
+                    owner:    Address::Account(self.args.keys.address),
+                    receiver: Address::Account(self.args.keys.address),
+                    data:     AdditionalData::new(vec![])?,
+                };
+
+                UpdateContractPayload {
+                    amount:       Amount::zero(),
+                    address:      self.contract_address,
+                    receive_name: OwnedReceiveName::new("cis2_wCCD.unwrap".into())?,
+                    message:      OwnedParameter::from_serial(&params)?,
+                }
+            }
+            _ => unreachable!(),
+        };
+
+        let tx = self.args.make_update_contract_transaction(
+            payload,
+            Energy::from(3500),
+            &mut self.nonce,
+        )?;
+        self.count += 1;
+
+        Ok(tx)
+    }
+}
+
+/// A generator that makes transactions that register dummy Web3 ID credentials.
+pub struct RegisterCredentialsGenerator {
+    args:             CommonArgs,
+    contract_address: ContractAddress,
+    nonce:            Nonce,
+    rng:              StdRng,
+}
+
+#[derive(concordium_std::Serial)]
+struct CredentialType {
+    #[concordium(size_length = 1)]
+    credential_type: String,
+}
+
+#[derive(concordium_std::Serial)]
+struct RegisterCredentialsInitParams {
+    issuer_metadata: MetadataUrl,
+    credential_type: CredentialType,
+    schema:          SchemaRef,
+    issuer_account:  Option<AccountAddress>,
+    issuer_key:      CredentialHolderId,
+    #[concordium(size_length = 1)]
+    revocation_keys: Vec<CredentialHolderId>,
+}
+
+#[derive(concordium_std::Serial)]
+struct CredentialInfo {
+    holder_id:        CredentialHolderId,
+    holder_revocable: bool,
+    valid_from:       Timestamp,
+    valid_until:      Option<Timestamp>,
+    metadata_url:     MetadataUrl,
+}
```

Comment:
There is a type called `CredentialInfo` already defined in cis4_types. That should be used.

Reply:
That type seems to have additional fields: `commitment` and `crendential_types`, won't that be a problem?

Reply:
Where do you see this, I'm talking about this type
```
#[derive(
    serde::Serialize, serde::Deserialize, contracts_common::Serialize, PartialEq, Eq, Clone, Debug,
)]
#[serde(rename_all = "camelCase")]
pub struct CredentialInfo {
    /// The holder's identifier.
    pub holder_id:        CredentialHolderId,
    /// Whether the holder is allowed to revoke the credential or not.
    pub holder_revocable: bool,
    /// The date from which the credential is considered valid.
    pub valid_from:       contracts_common::Timestamp,
    /// After this date, the credential becomes expired. `None` corresponds to a
    /// credential that cannot expire.
    pub valid_until:      Option<contracts_common::Timestamp>,
    /// Metadata URL of the credential.
    pub metadata_url:     MetadataUrl,
}
```